src/DGtal/io/readers/ITKReader.ih

```diff
@@ -64,7 +64,8 @@ namespace DGtal {
     const Domain& domain = dgtal_itk_image.domain();
 
     typedef ConstImageAdapter<DGtalITKImage, Domain, functors::Identity, Value, Functor> AdaptedImage;
-    const AdaptedImage adapted(dgtal_itk_image, domain, functors::Identity(), aFunctor);
+    const functors::Identity identityFunctor{};
```

Comment:
Well, I would have preferred to get the () removed instead of something that can be ambiguous for non-c11 dev. 
(Not sure to see why this new syntax enhance the construction.. I'll have to read the doc;))